linera-ethereum/src/client.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use ethers::{
+    prelude::{Http, Provider},
+    types::Log,
+};
+use ethers_core::types::{Address, BlockId, BlockNumber, Filter, U64};
+use ethers_middleware::Middleware;
+
+use crate::common::EthereumServiceError;
+
+pub async fn get_accounts_node(url: &str) -> Result<Vec<String>, EthereumServiceError> {
+    let provider = Provider::<Http>::try_from(url)?;
+    Ok(provider
+        .get_accounts()
+        .await?
+        .into_iter()
+        .map(|x| format!("{:?}", x))
+        .collect::<Vec<_>>())
+}
+
+pub async fn get_block_number_node(url: &str) -> Result<u64, EthereumServiceError> {
+    let provider = Provider::<Http>::try_from(url)?;
+    let block_number = provider.get_block_number().await?;
+    Ok(block_number.as_u64())
+}
+
+pub async fn get_balance_node(
+    url: &str,
+    address: &str,
+    block_nr: Option<u64>,
+) -> Result<[u64; 4], EthereumServiceError> {
```

Comment:
I do think that `linera-ethereum` should use and expose the Ethereum-specific types (rather than e.g. `[u64; 4]`). We should just keep them separate from Linera's core protocol through the oracle abstraction.